lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -968,19 +966,19 @@ def create_autoscaling_group(connection):
     metrics_collection = module.params.get('metrics_collection')
     metrics_granularity = module.params.get('metrics_granularity')
     metrics_list = module.params.get('metrics_list')
+    max_instance_lifetime = module.params.get('max_instance_lifetime')
+    if max_instance_lifetime is not None and not module.boto3_at_least('1.10.32'):
+        module.fail_json(
```

Comment:
Rather than running this test inside create_autoscaling_group I'd recommend running it in main() prior to any possible changes happening.